packages/chat-headless/src/ChatHeadless.ts

```diff
@@ -45,19 +207,19 @@ export class ChatHeadless {
   private isImpressionAnalyticEventSent = false;
 
   /**
-   * Constructs a new instance of the {@link ChatHeadless} class.
+   * Constructs a new instance of the {@link ChatHeadlessImpl} class.
    *
-   * @public
+   * @internal
    *
-   * @param config - The configuration for the {@link ChatHeadless} instance
+   * @param config - The configuration for the {@link ChatHeadlessImpl} instance
    * @param chatClient - An optional override for the default {@link ChatClient} instance
    */
   constructor(config: HeadlessConfig, chatClient?: ChatClient) {
```

Comment:
Can we update this constructor to just take in a core instance, which can be done in the provider layer, following the dependency injection approach. So:

```
//provider file
const defaultConfig: Partial<HeadlessConfig> = {
      saveToSessionStorage: true,
    };
    
provideChatHeadless(config) {
 const core: ChatCore = provideChatCore({ ...defaultConfig, ...config })
 return ChatHeadlessImpl(core)
}

provideChatHeadlessInternal(config, internalConfig) {
 const core: ChatCore = provideChatCoreInternal({ ...defaultConfig, ...config }, internalConfig)
 return ChatHeadlessImpl(core)
}

Would be easier for us to mock core as well I think